slsim/Sources/supernovae.py

```diff
@@ -81,3 +103,39 @@ def get_apparent_magnitude(self, time, band, zpsys="AB"):
         """
 
         return self.bandmag(band, zpsys, time)
+
+    def set_magnitudes(
+        self,
+        absmag,
+        abs_mag_band,
+        peak_apparent_mag,
+        peak_apparent_mag_band,
+        magsys,
+        cosmo=cosmology.FlatLambdaCDM(H0=70, Om0=0.3),
```

Comment:
this is dangerous setting a cosmology at input without the user knowing about it. The cosmology is really crucial for the peak magnitudes, and I would require the user of this definition to input it to enforce that the cosmology is being treated self-consistently throughout SLSim